federation/pkg/federation-controller/deployment/deploymentcontroller_test.go

```diff
@@ -117,6 +126,8 @@ func TestDeploymentController(t *testing.T) {
 			return cluster1Client, nil
 		case cluster2.Name:
 			return cluster2Client, nil
+		case cluster3.Name:
+			return cluster2Client, nil
```

Comment:
cluster3Client

It's a bit disconcerting that this test would pass with this typo.